dff/utils/benchmark/context_storage.py

```diff
@@ -0,0 +1,311 @@
+"""
+Context storage benchmarking
+----------------------------
+This module contains functions for context storages benchmarking.
+
+Basic usage::
+
+
+    from dff.utils.benchmark.context_storage import report
+    from dff.context_storages import context_storage_factory
+
+    storage = context_storage_factory("postgresql+asyncpg://postgres:pass@localhost:5432/test", table_name="benchmark")
+
+    report(storage)
+
+"""
+from uuid import uuid4
+from time import perf_counter
+import typing as tp
+
+from pympler import asizeof
+from tqdm.auto import tqdm
+
+try:
+    import matplotlib
+    from matplotlib.backends.backend_pdf import PdfPages
+    import matplotlib.pyplot as plt
+except ImportError:
+    matplotlib = None
+
+try:
+    import pandas
+except ImportError:
+    pandas = None
+
+try:
+    import polars
+except ImportError:
+    polars = None
+
+from dff.context_storages import DBContextStorage
+from dff.script import Context, Message
+
+
+def get_context_size(context: Context) -> int:
+    """Return size of a provided context."""
+    return asizeof.asizeof(context)
+
+
+def get_context(dialog_len: int, misc_len: int) -> Context:
+    """
+    Return a context with a given number of dialog turns and a given length of misc field.
+
+    Misc field is needed in case context storage reads only the most recent requests/responses.
+
+    Context size is approximately 1000 * dialog_len + 100 * misc_len bytes if dialog_len and misc_len > 100.
+    """
+    return Context(
+        labels={i: (f"flow_{i}", f"node_{i}") for i in range(dialog_len)},
+        requests={i: Message(text=f"request_{i}") for i in range(dialog_len)},
+        responses={i: Message(text=f"response_{i}") for i in range(dialog_len)},
+        misc={str(i): i for i in range(misc_len)},
+    )
```

Comment:
Maybe we should make message size random in length to reflect irregular nature of messages?